internal/scanner/scanner.go

```diff
@@ -151,26 +161,46 @@ func (s *Scanner) ScanClusterWideResources() error {
 	return s.reportStore.SaveAll()
 }
 
-func auditClusterResource(resource *resources.AuditableResources, resourcesFetcher *ResourcesFetcher, httpClient *http.Client, clusterReport *report.ClusterPolicyReport) {
+func auditClusterResource(resource *resources.AuditableResources, resourcesFetcher ResourcesFetcher, httpClient *http.Client, clusterReport, previousClusterReport *report.ClusterPolicyReport) {
 	for _, policy := range resource.Policies {
-		url, err := (*resourcesFetcher).GetPolicyServerURLRunningPolicy(context.Background(), policy)
+		url, err := resourcesFetcher.GetPolicyServerURLRunningPolicy(context.Background(), policy)
 		if err != nil {
 			// TODO what's the better thing to do here?
 			log.Error().Err(err)
 			continue
 		}
 		for _, resource := range resource.Resources {
+			if result, foundResult := previousClusterReport.GetResult(policy, resource); foundResult {
+				policyResourceVersion, ok := result.Properties[report.PropertyPolicyResourceVersion]
+				if ok && policyResourceVersion == policy.GetResourceVersion() {
+					// We have a result from the same policy version for the same resource instance.
+					// Skip the evaluation
```

Comment:
I think this isn't correct. We are looking at the policy version, but the resources may have changed, hence the PolicyReportResult for that policy may change too.

According to the RFC, https://github.com/kubewarden/rfc/blob/main/rfc/0011-audit-checks.md#step-5-attempt-to-reuse-previous-evaluation-results:

> If there's an entry, look at the evaluation timestamp of this entry. If policy update timestamp and pod instance update timestamp are older than the evaluation timestamp it means that nothing changed.

I think we need to look at the PolicyReportResult.timestamp, and use that to compare against the policy update timestamp and resource update timestamp, which have nothing to do with the resource version (generation).

Reply:
I'm aware that this change is not described in the RFC. However, I've spent some time trying to figuring out what would be the best "timestamp" field to be used. Let's use a [deployment](https://pkg.go.dev/k8s.io/api/apps/v1#Deployment) kind as example. There is just one timestamp, the [CreationTimestamp](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta). But this field has only the time when the resource is created. Not when it's updated. Thus, in the approached described in the RFC we could ended with  always reusing the old result. Even if the user changed some field in the deployment object. The same apply for the policy resource. 

The only field that I found that is updated on any change in the resourceVersion. See more in this [doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency):

> All Kubernetes resources have a "resourceVersion" field as part of their metadata. This resourceVersion is a string that identifies the internal version of an object that can be used by clients to determine when objects have changed

> The resourceVersion is changed by the server every time an object is modified. 

This makes sense if you consider the [ObjectReference](https://pkg.go.dev/k8s.io/api@v0.27.3/core/v1#ObjectReference) type. There is a field for resource version there. And the policy result type uses this references in the `Subjects` field. Thus, I believe that the use of the `ResourceVersion` is not incorrect. 

Am I missing something? If so, what is the timestamp field that you have in mind to be used in the comparison?

Reply:
JFYI: https://github.com/kubewarden/audit-scanner/pull/51#discussion_r1231090070

Reply:
I see. For a moment I though about the Status subresource, but that may not be present. Maybe it is worth to have a look on how others have implemented it.

Reply:
Let me check if I can find something... another option is to have a maximum time window witch the scanner can reuse result. For example, if the result is 1day old, reevaluate. This time window can be configurable by the user.

Reply:
I too think we have to use the `ResourceVersion` to understand if a resource has changed since the last time we looked at it